exchange/bidder_validate_bids.go

```diff
@@ -110,8 +110,11 @@ func validateBid(bid *pbsOrtbBid) (bool, error) {
 	if bid.bid.ImpID == "" {
 		return false, fmt.Errorf("Bid \"%s\" missing required field 'impid'", bid.bid.ID)
 	}
-	if bid.bid.Price <= 0.0 {
-		return false, fmt.Errorf("Bid \"%s\" does not contain a positive 'price'", bid.bid.ID)
+	if bid.bid.Price < 0.0 {
+		return false, fmt.Errorf("Bid \"%s\" does not contain a positive (or zero if there is a deal) 'price'", bid.bid.ID)
+	}
+	if bid.bid.Price <= 0.0 && bid.bid.DealID == "" {
```

Comment:
You should be able to have 'bid.bid.Price == 0.0 && bid.bid.DealID', I don't think it is possible for both 'x < 0.0' and 'x == 0.0' to be false yet 'x <= 0.0' be true.

Reply:
Ah yeah that makes sense. Will update!

Reply:
Just updated!